subcommand/get-consul-client-ca/command.go

```diff
@@ -0,0 +1,227 @@
+		return 1
+	}
+
+	if c.flagOutputFile == "" {
+		c.UI.Error(fmt.Sprintf("-output-file must be set"))
+		return 1
+	}
+
+	// create a logger
+	level := hclog.LevelFromString(c.flagLogLevel)
+	if level == hclog.NoLevel {
+		c.UI.Error(fmt.Sprintf("Unknown log level: %s", c.flagLogLevel))
+		return 1
+	}
+	logger := hclog.New(&hclog.LoggerOptions{
+		Level:  level,
+		Output: os.Stderr,
+	})
+
+	// create Consul client
+	consulClient, err := c.consulClient(logger)
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Error initializing Consul client: %s", err))
+		return 1
+	}
+
+	// Get the active CA root from Consul
+	// Wait until it gets a successful response
+	var activeRoot string
+	for activeRoot == "" {
+		caRoots, _, err := consulClient.Agent().ConnectCARoots(nil)
+		if err != nil {
+			logger.Info("Error retrieving CA roots from Consul", "err", err)
+			time.Sleep(1 * time.Second)
+			continue
+		}
+
+		activeRoot, err = getActiveRoot(caRoots)
+		if err != nil {
+			logger.Info("Could not get an active root", "err", err)
+			time.Sleep(1 * time.Second)
+			continue
+		}
+	}
+
+	err = ioutil.WriteFile(c.flagOutputFile, []byte(activeRoot), 0644)
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Error writing CA file: %s", err))
+		return 1
+	}
+
+	c.UI.Info(fmt.Sprintf("Successfully written Consul client CA to: %s", c.flagOutputFile))
+	return 0
+}
+
+func (c *Command) consulClient(logger hclog.Logger) (*api.Client, error) {
+	cfg := api.DefaultConfig()
+
+	// First, check if the server address is a cloud auto-join string.
+	// If it is, discover server addresses through the cloud provider.
+	if strings.Contains(c.flagServerAddr, "provider=") {
+		disco, err := c.newDiscover()
+		if err != nil {
+			return nil, err
+		}
+		logger.Debug("using cloud auto-join with", c.flagServerAddr)
+		servers, err := disco.Addrs(c.flagServerAddr, logger.StandardLogger(&hclog.StandardLoggerOptions{
+			InferLevels: true,
+		}))
+		if err != nil {
+			return nil, err
+		}
+
+		// check if we discovered any servers
+		if len(servers) == 0 {
+			return nil, fmt.Errorf("could not discover any Consul servers with %q", c.flagServerAddr)
+		}
+
+		logger.Debug("discovered servers", strings.Join(servers, " "))
+
+		// Pick the first server from the list,
+		// ignoring the port since we need to use HTTP API
+		firstServer := strings.SplitN(servers[0], ":", 2)[0]
+		cfg.Address = fmt.Sprintf("%s:8501", firstServer)
+		cfg.Scheme = "https"
+	} else {
+		// check if the server URL is missing a port
+		host := strings.TrimPrefix(c.flagServerAddr, "https://")
+		host = strings.TrimPrefix(c.flagServerAddr, "http://")
+		parts := strings.SplitN(host, ":", 2)
+
+		// Use the default HTTPS port if port is missing.
+		// Otherwise, use the address the user has provided.
+		if len(parts) == 1 {
+			cfg.Address = fmt.Sprintf("%s:8501", c.flagServerAddr)
+			cfg.Scheme = "https"
+		} else {
+			cfg.Address = c.flagServerAddr
+		}
+	}
+
+	if c.flagCAFile != "" {
```

Comment:
Good catch. I actually think it makes sense to not overwrite it since that is what [consul flags do](https://github.com/hashicorp/consul/blob/382d33bb7e21ea62d53089a26389ad375c297829/command/flags/http.go#L131-L150). I think it's good to keep this behavior consistent across, even though this is a separate binary.